src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -286,6 +286,18 @@ def initialize_engines(
             self.cache_support_enabled and self.enable_multitoken_prefill
         ) or not self.cache_support_enabled:
 
+            # input_ids_length for the multitoken engine is either:
+            # - the prompt_processing_sequence_length if the cache support is enabled
+            #   (the prompt is processed sequentially at predefined processing length)
+            # - the full sequence_length if the cache support is disabled
+            #   (the prompt is processed in a single pass, prompts length is fixed at
+            #   sequence_length)
```

Comment:
Will we have tests soon to verify the different workflows? I know you're working on adding in testing so unit tests for these different workflows will be good to add.

LGTM otherwise.